### ISSUE TITLE
Error when searching subtitle using addic7ed

### ISSUE THREAD

**Issue by Olivier6767** [pymedusa/Medusa#3166]
Branch: develop
Commit: 716605fcd4009c94532bdb090bfa76b8d56af11f
Version: 0.1.16-dev0
Database: 44.8

Hello,
I'm having an issue every time there is a search for subtitles. I have enbaled Addic7ed and TVSubTitles and put my account information in the plugin settings.

For Addic7ed, I always get the error message below:

2017-10-19 13:05:46 WARNING  Thread-16 :: [716605f] Unexpected error in provider u'addic7ed'
Traceback (most recent call last):
  File "/home/Services/cinema/medusa/Medusa/ext/subliminal/core.py", line 119, in list_subtitles_provider
    return self[provider].list_subtitles(video, provider_languages)
  File "/home/Services/cinema/medusa/Medusa/ext/subliminal/providers/addic7ed.py", line 275, in list_subtitles
    return [s for s in self.query(video.series, video.season, video.year)
  File "/home/Services/cinema/medusa/Medusa/ext/subliminal/providers/addic7ed.py", line 240, in query
    raise TooManyRequests()
TooManyRequests

**Comment by sharkykh** [comment]
Addic7ed.com actively blocks Subliminal, which is the subtitle downloader Medusa uses.

**Comment by medariox** [comment]
Choose another subtitles provider or try to use multiple providers to level out the load. Sadly there is nothing we can do about this.

**Comment by Olivier6767** [comment]
Sorry for the dealyed answer, but I made multiple tests and I don't think the issue is with the web site.
I restarted Sickrage (which I was using before switching to Medusa) and manually triggered search for the subtitles I was missing in Medusa.
They were all found and downloaded within seconds. I had only enbaled the addic7ed provider.

Currently in Medusa, I haved enbaled the main 3 providers (addic7ed, tvsubtitles and opensubtitles).
- addic7ed fails with TooManyRequests - but it works in Sickrage
- tvsubtitles and opensubtitles don't have the french subtitles yet

addic7ed is by far the best service. It seems that the TooManyRequests message is not correct.
When checking my account on addic7ed, I only used a couple of API search out of the 40 I'm allowed, and again, this provider works perfectly in Sickrage.
I hope we can fix it in Medusa, as currently none of my subtitles are being downloaded...

Thanks for your help,
Olivier

**Comment by sharkykh** [comment]
As far as I know, Subliminal doesn't use the site's API. And when I changed the User Agent defined inside Subliminal, it started working again.

That said, it should not stop searching when that exception is thrown, I think. @medariox

**Comment by medariox** [comment]
Sorry if it seemed rude to close the issue, but I really think there is nothing we can do about this.

@sharkykh is right, Subliminal doesn't use any API for addic7ed, I don't know what the "API search" is supposed to represent in your case - not even sure they actually have an API. SR uses exactly the same software (Subliminal) to download the subtitles, it seems unlikely that it would work over there unless they override the User Agent (which isn't an acceptable solution IMO).

Apparently, [Addic7ed is blocking Subliminal](https://github.com/Diaoul/subliminal/issues/788#issuecomment-318861428) and that's why Subliminal raises `TooManyRequests()`. At the end of the day, it's their site, they have the right to exclude automated downloaders like Subliminal if they wish to.

We are open to suggestions to solve this issue, I just don't see any doable solution yet.

@sharkykh It should definitely continue searching with the other providers. If the search stops completely, it's a bug.

**Comment by sharkykh** [comment]
> @sharkykh It should definitely continue searching with the other providers. If the search stops completely, it's a bug.

Sorry, I just assumed. I didn't actually check.

**Comment by duramato** [comment]
> I only used a couple of API search out of the 40 I'm allowed...

Aren't you confusing it with allowed downloads? There's no mention on api search.

![image](https://pls.gtfo.accountant/bMCSA.png)

**Comment by Olivier6767** [comment]
Thanks for the insights.
I've tried using Subliminal directly and was able to download several subtitles. Eventually I started getting the Too Many Requests message also in Subliminal. 
I'm still well under the 40 download limit, so I suspect there is a limit on the number of searches maybe, even though it does not explain why it works for one tool and not the other... 
I've tried changing the User-Agent to see if this would work, but I still get the error message. 

I've contacted the site admins to ask about this. I will post their answers here as soon as I get it.

**Comment by Olivier6767** [comment]
I never got an answer from the site admins. 
Closing this topic for now.

**Comment by Olivier6767** [comment]
Hello,
After spending more time debugging this issue, here is what I found out:
I have many shows where I had downloaded the first seasons in local language (french). I did this by adding a **required words (french)** in the Advanced season settings.
Once I had watched one season, I **marked** all the episodes for this season as **Archived**.
For the latest seasons, since the French version is usually not available upon release, I **removed the required keyword** French and I **enabled Search subtitles** in the Main season settings.

What I found out by reviewing the debug logs, is that Medusa still wants to download all missing subtitles for old seasons marked as Archived. This represents a huge list of subtitles.
I believe that this is what causes the **Too Many Requests** error message from the addic7ed subtitle provider.

Is this a normal behaviour? 
Is there a setting that waould avoid Medusa trying to download old subtitles (for episodes marked as Archived)?

Thanks.
